### ISSUE TITLE
Xcode 7 compiler warnings

### ISSUE THREAD

**Issue by Blog404DE** [hackiftekhar/IQKeyboardManager#291]
After compiling the objc-demo i've seen some (new) warnings:

***IQKeyboardReturnKeyHandler.m*** 
```
IQKeyBoardManager/IQKeyboardReturnKeyHandler.m:51:17: Method override for the designated initializer of the superclass '-init' not found
```

***IQTitleBarButtonItem.m***
```
IQKeyBoardManager/IQToolbar/IQTitleBarButtonItem.m:42:19: Designated initializer invoked a non-designated initializer
IQKeyBoardManager/IQToolbar/IQTitleBarButtonItem.m:42:19: Designated initializer invoked a non-designated initializer
IQKeyBoardManager/IQToolbar/IQTitleBarButtonItem.m:40:1: Designated initializer missing a 'super' call to a designated initializer of the super class
IQKeyBoardManager/IQToolbar/IQTitleBarButtonItem.m:33:17: Method override for the designated initializer of the superclass '-init' not found
IQKeyBoardManager/IQToolbar/IQTitleBarButtonItem.m:33:17: Method override for the designated initializer of the superclass '-initWithCoder:' not found
```
Maybe somebody have the time to sort this warnings out and fix it :+1:

**Comment by Piero87** [comment]
i have the same warning, any fix please?

**Comment by peng90** [comment]
I'm getting same warnings in Xcode 7 GM.

**Comment by RobertNNms** [comment]
Hi,

Add this in IQTitleBarButtonItem.h :

- (instancetype)initWithFrame:(CGRect)frame title:(NSString *)title;
- (instancetype)init NS_UNAVAILABLE;
- (instancetype)initWithCoder:(NSCoder *)aDecoder NS_UNAVAILABLE;

Regards
Robert

**Comment by iOSCuriosity** [comment]
@hackiftekhar Are you working on this enhancement right now? Do we expect any official update for this warnings from your side?
@RobertNNms This may resolve the warning temporary (for those who uses podfile).. anyhow thanks for the solution.!! I'm gonna use it..

**Comment by Piero87** [comment]
@RobertNNms  that fix is not enough, i have also edit this line:

    -(instancetype)initWithViewController:(UIViewController*)controller NS_DESIGNATED_INITIALIZER;

to this:
    -(instancetype)initWithViewController:(UIViewController*)controller;

in IQKeyboardReturnKeyHandler, it's right?

**Comment by hackiftekhar** [comment]
Fixed now ac6cbcb0aec00d51a7071f05c6e1a3f651900306.
